### PR TITLE
Update function calling examples to Python SDK

### PR DESCRIPTION
**Pull request by teomusatoiu**
## Summary

Changed the cookbook to use the Python SDK instead of the requests library.

## Motivation

The page of the cookbook was outdated and had to reflect the newest capabilities.

---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [ ] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [ ] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [ ] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [ ] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [ ] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [ ] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [ ] Correctness: The information I include is correct and all of my code executes successfully.
  - [ ] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

**Review by ibigio — APPROVED** [review]
Thanks for these updates!

File: examples/How_to_call_functions_with_chat_models.ipynb [hunk 1]
```diff
@@ -12,7 +12,7 @@
...
+    "Within the `tools` parameter, if the `functions` parameter is provided then by default the model will decide when it is appropriate to use one of the functions. The API can be forced to use a specific function by setting the `tool_choice` parameter to `{\"name\": \"<insert-function-name>\"}`. The API can also be forced to not use any function by setting the `tool_choice` parameter to `\"none\"`. If a function is used, the output will contain `\"finish_reason\": \"function_call\"` in the response, as well as a `tool_choice` object that has the name of the function and the generated function arguments.\n",
```
Comment (ARajgor) [inline examples/How_to_call_functions_with_chat_models.ipynb:15]: Docs error. tool_choice doesn't accept `{"name" : ""}` it required `{"type": "function", "function": {"name": "my_function"}}`.
Read the [docs](https://platform.openai.com/docs/api-reference/chat/create#chat-create-tool_choice)

File: examples/How_to_call_functions_for_knowledge_retrieval.ipynb
Comment (ARajgor) [inline examples/How_to_call_functions_for_knowledge_retrieval.ipynb:1]: use tools and tools_call instead of depricated function_call.

**Review by ARajgor — COMMENTED** [review]
@teomusatoiu  You did a great job on migrating to python SDK but both files contain some problems. You change the function parameter. according to [Docs](https://platform.openai.com/docs/api-reference/chat/create#chat-create-function_call) function_call is deprecated and replaced by tools and tools_choices.

File: examples/How_to_call_functions_with_chat_models.ipynb (hunk 1 above)
Comment (teomusatoiu) [reply examples/How_to_call_functions_with_chat_models.ipynb:15]: Hi @ARajgor thank you for noticing the docs flaw. I just opened[ this PR](https://github.com/openai/openai-cookbook/pull/1030) that should solve this issue.

Comment (ARajgor) [reply examples/How_to_call_functions_with_chat_models.ipynb:15]: We have already migrated many files to V1 API. we had a big PR request that got merged a few days ago. So check both of the files regarding old codes/deprecated APIs.

**Pull request merged**